### PR TITLE
Upgrades o-comment, replaces dom-delegate with ftdomdelegate. Fixes #39.

### LINKED ISSUE #39: Switch dom-delegate for ftdomdelegate -- remove alias.
`ftdomdelegate` and `dom-delegate` are the same except for their name. Matt recently tidied up `ftdomdelegate` and published as `ftdomdelegate@^3.0.0` to npm, deprecating `dom-delegate`:
https://github.com/Financial-Times/ftdomdelegate/pull/93
https://github.com/Financial-Times/ftdomdelegate/releases/tag/v3.0.0

So we can  finally just include `ftdomdelegate` and remove `dom-delegate` and aliases. :)
https://github.com/Financial-Times/g-components/blob/7cb69e7c65f225b6ca8dc9975a7fd5b9c9bb3120/package.json#L54
https://github.com/Financial-Times/g-components/blob/1562b586c550c8a25a3861bd246d6af962f8fbea/.storybook/webpack.config.js#L19

You'll want to bump to the new o-comments major `o-comments@^5.0.1` too, as it used to rely directly on `dom-delegate`.
https://github.com/Financial-Times/o-comments/pull/29

Thanks to Matt H for this one. Hope the change helps simplify your codebase a little :) Let me know if you have any questions.

### PR DESCRIPTION
**Pull request by aendrew**
Build succeeds locally.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "o-forms": "^5.8.2"
   },
   "resolutions": {
-    "o-ads": "^10.0.1"
+    "o-ads": "^10.0.1",
```
Comment (notlee) [inline bower.json]: Is the o-ads resolution needed? Looks like o-ads v11 moves from CJS to ESM, and I just released `o-ads@^11.0.2` which includes `ftdomdelegate@^3.0.0`.

File: bower.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
...
+    "o-comments": "^5"
```
Comment (notlee) [inline bower.json]: How come there's a `o-comments` resolution here? Have I missed a dependent on `o-comments` which needs updating?

**Comment by notlee** [comment]
Nice one for jumping on the upgrade :)

File: bower.json (hunk 2 above)
Comment (aendrew) [reply bower.json]: I'm not sure why it was wanting a resolution; I've removed both of those and done a bower install without issue.

**Pull request merged**